src/oapv_metadata.c

```diff
@@ -180,6 +180,97 @@ static void meta_free_md(oapv_md_t *md)
     }
 }
 
+int oapvm_write_mdcv(oapvm_payload_mdcv_t *mdcv, void *data, int *size)
+{
+    int i, t;
+    u32 tu32;
+    oapv_bs_t bs;
+    oapv_bsw_init(&bs, data, 24, NULL); // MDCV payload has 24 bytes
+
+    for(i = 0; i < 3; i++) {
+        t = mdcv->primary_chromaticity_x[i];
+        oapv_assert_rv(t >=0 && t < (2^16), OAPV_ERR_INVALID_ARGUMENT);
```

Comment:
The operator ^ denotes XOR (a bitwise operation), not exponentiation. In the code, 2^16 is used, resulting in 2 XOR 16 = 18, not 65536. The correct syntax for exponentiation is pow(2, 16)  or manually calculating it as 1 << 16
This causes the condition in an assert statement is not met, for values >= 18 [master-display=G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(10000000,1)]
Please change (2^16) to (1<<16)

Reply:
Mistake, It is changed to 0xFFFF